Lib/fontbakery/profiles/gdef.py

```diff
@@ -0,0 +1,153 @@
+from fontbakery.callable import check
+from fontbakery.checkrunner import FAIL, PASS, WARN
+from fontbakery.message import Message
+from fontbakery.utils import pretty_print_list
+
+# used to inform get_module_profile whether and how to create a profile
+from fontbakery.fonts_profile import profile_factory # NOQA pylint: disable=unused-import
+
+from fontTools import unicodedata
+from fontbakery.constants import GDEF_MARK_GLYPH_CLASS
+
+
+def _is_non_mark_char(charcode):
+  category = unicodedata.category(chr(charcode))
+  if category.startswith("C"):
+    # skip control characters
+    return None
+  elif not category.startswith("M"):
+    return True
+  return False
+
+
+def _get_mark_class_glyphs(ttFont):
+  class_defs = ttFont["GDEF"].table.GlyphClassDef.classDefs.items()
+  return {name for (name, value) in class_defs
+          if value == GDEF_MARK_GLYPH_CLASS}
+
+
+@ check(
+  id = 'com.google.fonts/check/gdef_spacing_marks',
+  rationale = """
+    Glyphs in the GDEF mark glyph class should be non-spacing.
+    Spacing glyphs in the GDEF mark glyph class may have incorrect
+    anchor positioning that was only intended for building composite glyphs
+    during design.
+  """
+)
+def com_google_fonts_check_gdef_spacing_marks(ttFont):
+  """Check mark characters are in GDEF mark glyph class)"""
+  if "GDEF" in ttFont and ttFont["GDEF"].table.GlyphClassDef:
+    spacing_glyphs = {name for (name, value) in ttFont["hmtx"].metrics.items()
+                      if value}
```

Comment:
This will consider any glyph a spacing glyph, I suppose it should have been `if value[0]`.